frontend/src/metabase/dashboard/hoc/controls/use-dashboard-refresh-period.ts

```diff
@@ -0,0 +1,68 @@
+import { useInterval } from "@mantine/hooks";
+import { useRef, useState } from "react";
+import { useUnmount } from "react-use";
+
+import * as MetabaseAnalytics from "metabase/lib/analytics";
+
+export type RefreshPeriod = number | null;
+
+export type DashboardRefreshPeriodControls = {
+  refreshPeriod: RefreshPeriod;
+  onRefreshPeriodChange: (newPeriod: RefreshPeriod) => void;
+  setRefreshElapsedHook: (
+    hook: DashboardRefreshPeriodControls["onRefreshPeriodChange"],
+  ) => void;
+};
+
+const TICK_PERIOD = 1; // seconds
+export const useDashboardRefreshPeriod = ({
+  onRefresh,
+}: {
+  onRefresh: () => void;
+}): DashboardRefreshPeriodControls => {
+  const [period, setPeriod] = useState<number | null>(null);
+  const elapsedHook = useRef<((elapsed: number | null) => void) | null>(null);
+  const elapsed = useRef<number | null>(0);
+
+  const setRefreshElapsedHook = (hook: (elapsed: number | null) => void) => {
+    elapsedHook.current = hook;
+  };
+
+  // const setPeriod = (newPeriod: number | null) => {
+  //   period.current = newPeriod;
+  // };
+
+  const { start, stop } = useInterval(() => {
```

Comment:
I changed this from the manual interval logic to this Mantine hook, which handles most of the pain for us and reduces our complexity.

I also used Mantine's `useInterval` hook instead of `react-use` since it provides a little more control over when we can start and stop the interval.